[FLINK-3719][web frontend] Moving the barrier between graph and stats

mushketyk opened the pull request:
Added a moving split barrier between graph and stats.
It looks like this:
![split](https://cloud.githubusercontent.com/assets/592286/18220366/19e49bd0-7169-11e6-949c-7839033101bb.png)

Bottom part becomes scrollable if a user makes it too small, the top part is never scrollable.
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)
- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added
- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

StephanEwen commented:
This is a useful extension.
I have pinged someone with more web UI experience to take a look at how it is implemented.

Concerning licenses: Where are the png images for the sliders from? Were they self-created?
In the case of other binary files, we usually left a text file in the directory stating the origin and license of the files.

mushketyk commented:
Hi @StephanEwen 

All images are coming from this repo: https://github.com/nathancahill/Split.js
It has a very permissive license: https://github.com/nathancahill/Split.js/blob/master/LICENSE.txt

StephanEwen commented:
The license is MIT, so that is fine.
Can you update the LICENSE file with the dependency?

StephanEwen commented:
Piotr (who wrote most of the web ui) also wants to leave some comments. Let's wait for him.

iampeter commented:
@mushketyk thanks for this feature, very useful indeed.

I'd like to ask you to reconsider some things:
1. First of all, I think it would be good to wrap split.js in an Angular directive
2. I think the `style` attribute could be replaced by CSS
3. I think there is a typo in the port numer (80801)

Let me know your thoughts.

Thanks,
Piotr

mushketyk commented:
Hi @iampeter 
Thank you for your review.
I'll update the code according to your suggestions.

Best regards,
Ivan.

mushketyk commented:
Hi @iampeter @StephanEwen 

I've fixed the PR according to your review. I've added new license, introduced a new directive, and updated CSS.

Could you please review it once again?

iampeter commented:
Thanks, that's better @mushketyk, but I think would be even better if you did not identify the elements explicitly by ids ('#canvas', '#job-panel'), but have the directive contain the two splitable parts, i.e.:

```
<split>
  <div>Upper part</div>
  <div>Lower part</div>
</split>
```

and just use the first and second child to identify them. This way the directive would be reusable and also could be used in other parts of the dashboard.

mushketyk commented:
Hi @iampeter,

Sorry for the delay in response. I'll take a look at this on Sunday.

mushketyk commented:
Hi @iampeter 
I've updated the PR according to your comments. Could you please review it again?

Best regards,
Ivan.

iampeter commented on flink-runtime-web/web-dashboard/app/scripts/modules/jobs/jobs.dir.coffee:
```diff
@@ -168,14 +168,15 @@ angular.module('flinkApp')
+      getId = (elem) -> "#" + elem.id
+      id1 = getId tElem.children().get(0)
+      id2 = getId tElem.children().get(1)
+      Split([id1, id2], (
```
I think that since Split.js accepts HTML elements, we could just write:
```
compile: (tElem, tAttrs) ->
   Split(tElem.children(), (
   ...
```

mushketyk replied:
Good point. I'll check if this works.

mushketyk commented:
Hi @iampeter 

I've updated the PR according to your review.

mushketyk commented:
@iampeter Is there anything else that I should change?

iampeter commented:
@mushketyk do make sure that the generated files are up-to-date (if they're not)

greghogan commented:
Merging this ...
